torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -376,8 +376,15 @@ TensorPipeAgent::TensorPipeAgent(
           tensorpipe::ContextOptions().name(workerInfo_.name_))),
       rankToNameStore_("names", store),
       nameToAddressStore_("addrs", store),
-      shutdownStore_("shutdown", store),
-      worldSize_(worldSize) {
+      shutdownStore_("shutdown", store) {
+  if (worldSize.has_value()) {
+    worldSize_ = worldSize.value();
+    isStaticGroup_ = true;
+  } else {
+    // TODO: get current world size from store
+    isStaticGroup_ = false;
```

Comment:
I assume all processes must all set `isStaticGroup_` to the same value? If so, let's also write this to store as well. If any process detected it's local `isStaticGroup_` is different from the one in store, it should error out.

Reply:
Created a new method `checkAndSetStaticGroup()` to handle this logic during initialization